joblib/test/test_parallel.py

```diff
@@ -152,6 +152,23 @@ def test_effective_n_jobs():
     assert effective_n_jobs() > 0
 
 
+@pytest.mark.parametrize(
+    "backend_n_jobs, expected_n_jobs",
+    [(3, 3), (-1, effective_n_jobs(n_jobs=-1)), (None, 1)],
```

Comment:
```suggestion
    [(3, 3), (-1, cpu_count()), (None, 1)],
```

would be more explicit.

Reply:
yes it is less weird because it is the the function that we try to test.

Reply:
I also find it more explicit but this could lead to border effects, for instance as we discuss to only use the number of physical cores by calling `cpu_count(physical=True)` at some point or something like this.
The `effective_n_jobs` option feels safer as it does not depends on the implementation in the backend.

Reply:
ok